src/index.d.ts

```diff
@@ -0,0 +1,180 @@
+
+}
+
+export class Operation extends Brush {
+
+  isOperation: boolean;
+  // operation
+  markUpdated(): void;
+  isDirty(): boolean;
+  insertBefore( brush: Brush );
+  insertAfter( brush: Brush );
+
+}
+
+export class OperationGroup extends Group {
+
+  isOperatioinGroup: boolean;
+  markUpdated(): void;
+  isDirty(): boolean;
+
+}
+
+export class CullableTriangle extends Triangle {
+
+  initFrom( other: Triangle );
+  updateSide( plane: Plane, triangle: Triangle, coplanarIndex: number );
+
+}
+
+export class TrianglePool {
+
+  getTriangle(): Triangle;
+  clear(): void;
+  reset(): void;
+
+}
+
+export class TriangleSplitter {
+
+  trianglePool: TrianglePool;
+  triangles: Triangle[];
+  normal: Vector3;
+
+  initialize( tri: Triangle );
+  splitByTriangle( triangle: Triangle );
+  splitByPlane( plane: Plane, triangle: Triangle, coplanarIndex: number );
+  reset(): void;
+
+}
+
+export class HalfEdgeMap {
+
+  constructor( geometry?: BufferGeometry );
+  getSiblingTriangleIndex( triIndex: number, edgeIndex: number ): number;
+  getSiblingEdgeIndex( triIndex: number, edgeIndex: number ): number;
+  updateFrom( geomtry: BufferGeometry ): void;
+
+}
+
+export class GridMaterial extends MeshPhongMaterial {
+
+  enableGrid: boolean;
+
+  onBeforeCompile( shader: ShaderMaterial ): void;
+  customProgramCacheKey(): boolean;
+
+}
+
+export function getTriangleDefinitions( ...triangles: Triangle[] ): String[];
+
+export function logTriangleDefinitions( ...triangles: Triangle[] ): void;
+
+export function generateRandomTriangleColors( geometry: BufferGeometry ): void;
+
+export class TriangleSetHelper extends Group {
+
+  constructor( triangles?: Triangle[] );
+
+  setTriangles( triangles: Triangle[] );
+
+}
+
+export class EdgesHelper extends LineSegments {
+
+  constructor( edges?: EdgesGeometry );
+
+  setEdges( edges: EdgesGeometry ): void;
+
+}
+
+export class TriangleIntersectData {
+
+  intersects: Object;
+  triangle: Triangle;
+
+  constructor( tri: Triangle );
+  addTriangle( index: number, tri: Triangle ): void;
+  getIntersectArray(): Array;
+
+}
+
+export class TriangleIntersectionSets {
+
+  addTriangleIntersection( ia: number, tribA: Triangle, ib: number, triB: Triangle ): void;
+  getTrianglesAsArray( id?: number ): Array;
+  getTriangleIndices(): Array;
+  getIntersectionIndices( id: number );
+  getIntersectionsAsArray( id?: number, id2: number ): Array;
+
+}
+
+export class OperationDebugData {
+
+  intersectionEdges: EdgesGeometry[];
+
+  addIntersectingTriangles( ia: number, triA: Triangle, ib: number, triB: Triangle ): void;
+  addEdge( edge: EdgesGeometry ): void;
+  reset(): void;
+
+}
+
+export class PointsHelper extends InstancedMesh {
+
+  constructor( count?: number, points?: Vector3[] );
+
+  setPoints( points: Vector3 ): void;
+
+}
+
+// export class HalfEdgeHelper extends EdgesHelper {
+
+//   constructor(geometry?: BufferGeometry, halfEdges?: );
+//   setHalfEdges(geometry: BufferGeometry, halfEdges: ): void;
+
+// }
```

Comment:
Why is this one commented out?

Reply:
Because I'm not sure which is the correct type for halfEdges param, THREE.HalfEdge?

Reply:
This half edge map build internally is the one to pass in:

https://github.com/gkjohnson/three-bvh-csg/blob/main/src/core/HalfEdgeMap.js